.gitlab-ci.yml

```diff
@@ -1,8 +1,92 @@
+    - export NVM_DIR="$([ -z "${XDG_CONFIG_HOME-}" ] && printf %s "${HOME}/.nvm" || printf %s "${XDG_CONFIG_HOME}/nvm")" && [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+    - nvm install $NODE_VERSION
+    - wget -O /tmp/chrome.deb https://dl.google.com/linux/chrome/deb/pool/main/g/google-chrome-stable/google-chrome-stable_99.0.4844.82-1_amd64.deb
+    - apt-get install -y /tmp/chrome.deb
+    - cd lib/jsorolla && npm install && cd ../.. && npm install
+    - npm install -g grunt-cli
+    - apt-get install xvfb && Xvfb :99 & export DISPLAY=:99
+  script:
+    - grunt run-all-tests --env=staging
+  environment:
+    name: test-env
+  only:
+    - master
+    - tags
+
+# Not executed, parent job definition
+.deploy-website:
+  stage: deploy
+  image: ubuntu:20.04
+  environment:
+    name: $ENVIRONMENT_NAME
+  before_script:
+    - apt-get update
+    - apt-get upgrade -y
+    - apt-get install -y tzdata
+    - apt-get install -y wget
+    - apt-get install -y git
+    - apt-get install -y zip
+    - apt-get install -y rsync
+    - git submodule update --init
+    - wget -qO- https://raw.githubusercontent.com/nvm-sh/nvm/v$NVM_VERSION/install.sh | bash
+    - export NVM_DIR="$([ -z "${XDG_CONFIG_HOME-}" ] && printf %s "${HOME}/.nvm" || printf %s "${XDG_CONFIG_HOME}/nvm")" && [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+    - nvm install $NODE_VERSION
+    - cd lib/jsorolla && npm install && cd ../.. && npm install
+    - npm install -g grunt-cli
   script:
-    - echo "Test CI/CD script"
+    - grunt --env=$ENVIRONMENT_NAME
+    - ARTIFACT_NAME=$(basename $(ls -d build/*$ENVIRONMENT_NAME))
+    # Start ssh-agent and set identity then copy the file to website mount point and a compressed version to cluster
+    # See https://docs.gitlab.com/ee/ci/ssh_keys/
+    - apt-get -y install openssh-client
+    - eval "$(ssh-agent)"
+    - echo "$VAR_ADM_PRIVATE_KEY" | tr -d '\r' | ssh-add -
+    - rsync -av -q -e "ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null" build/${ARTIFACT_NAME}/* ${WEBSITE_DEPLOY_USER}@${CLUSTER_LOGIN_NODE}:/nfs/public/rw/variation/projects/eva/$WEBSITE_DEPLOY_ENVIRONMENT/www/
+    - echo "$EVA_ETL_PRIVATE_KEY" | tr -d '\r' | ssh-add -
+    - cd build && zip -q -r ${ARTIFACT_NAME}.zip ${ARTIFACT_NAME}
+    - scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null ${ARTIFACT_NAME}.zip ${CLUSTER_USER}@${CLUSTER_LOGIN_NODE}:${EVA_SOFTWARE_PATH}/eva-web/
+  dependencies:
+    - test
+
+deploy-website-internal:
+  extends: .deploy-website
+  variables:
+    ENVIRONMENT_NAME: dev
+    WEBSITE_DEPLOY_ENVIRONMENT: dev
+  only:
+    - master
+
+deploy-website-development:
+  extends: .deploy-website
+  variables:
+    ENVIRONMENT_NAME: staging
+    WEBSITE_DEPLOY_ENVIRONMENT: stage
+  only:
+    - master
+
+deploy-website-production:
+  extends: .deploy-website
+  variables:
+    ENVIRONMENT_NAME: prod
+    WEBSITE_DEPLOY_ENVIRONMENT: prod
```

Comment:
Production deployment alone has an extra deploy script to run, see the last couple steps on [this page](https://www.ebi.ac.uk/seqdb/confluence/display/VAR/Web+Servers#WebServers-Productionservers).  We could add it to the CI pipeline or maybe leave it as a manual step for production deployment.

Reply:
Thanks for catching this..  have added it in the ci pipeline